src/main/kotlin/org/rust/ide/refactoring/move/RsMoveFilesOrDirectoriesConflictsDetector.kt

```diff
@@ -0,0 +1,123 @@
+        // - items in movedFile can have `pub(in ...)` visibility,
+        //   but we will update them in performRefactoring
+
+        shouldMakeMovedFileModDeclarationPublic = false
+        for (usage in usages) {
+            val usageElement = usage.element as? RsPath ?: continue
+            val usageMod = usageElement.containingMod
+
+            // TODO: better support for reexports
+            for (superMod in newParentMod.superMods.asReversed()) {
+                if (!superMod.isVisibleFrom(usageMod)) {
+                    addVisibilityConflict(conflicts, usageElement, movedFile)
+                    break
+                }
+            }
+
+            if (!usageMod.superMods.contains(newParentMod)) {
+                shouldMakeMovedFileModDeclarationPublic = true
+            }
+        }
+    }
+
+    private fun detectOutsideReferencesVisibilityProblems(conflicts: MultiMap<PsiElement, String>) {
+        for ((path, target) in outsideReferencesMap) {
+            val isInsideVisRestriction = path.parents().first { it !is RsPath } is RsVisRestriction
+            if (isInsideVisRestriction || target.crateRoot != movedFile.crateRoot) continue
+
+            if (path.hasOnlySuperSegments()) {
+                val pathParent = path.parent
+                if (pathParent is RsPath && pathParent.hasOnlySuperSegments()) continue
+
+                val targetMod = target as? RsMod ?: continue
+                for (mod in targetMod.superMods) {
+                    if (!mod.isVisibleFrom(newParentMod)) {
+                        addVisibilityConflict(conflicts, path, target)
+                        break
+                    }
+                }
+            } else {
+                checkVisibility(conflicts, path)
+            }
+        }
+
+        detectPrivateStructFieldOutsideReferences(conflicts)
+    }
+
+    private fun detectPrivateStructFieldOutsideReferences(conflicts: MultiMap<PsiElement, String>) {
+        fun checkStructFieldVisibility(fieldReference: RsMandatoryReferenceElement) {
+            val field = fieldReference.reference.resolve() ?: return
+            if (!field.isInsideModSubtree(movedFile)) checkVisibility(conflicts, fieldReference)
+        }
+        for (dotExpr in movedFile.descendantsOfType<RsDotExpr>()) {
+            val fieldReference = dotExpr.fieldLookup ?: dotExpr.methodCall ?: continue
+            checkStructFieldVisibility(fieldReference)
+        }
+        for (fieldReference in movedFile.descendantsOfType<RsStructLiteralField>()) {
+            checkStructFieldVisibility(fieldReference)
+        }
+        for (patField in movedFile.descendantsOfType<RsPatField>()) {
+            val patBinding = patField.patBinding ?: continue
+            checkStructFieldVisibility(patBinding)
+        }
+
+        for (patTupleStruct in movedFile.descendantsOfType<RsPatTupleStruct>()) {
+            val struct = patTupleStruct.path.reference?.resolve() as? RsStructItem ?: continue
```

Comment:
Thank you, I didn't know about `deepResolve`!
Looks like it is not needed in this case, because [type aliases can't be used in destructuring tuple struct](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=26237db6cac3751e74b3bf3b667a8b68)